pages/api/wallets/[viewWalletPubkey].ts

```diff
@@ -79,17 +89,38 @@ export default async function handler(
         const fanoutObj = await validateWallet(cluster, viewWalletPubkey)
         // If exits
         if (fanoutObj) {
-          // Fetch SPL Details (if accept)
-          // TODO
-
+          // Fetch SPL Details and Update wallet if it accept SPL token
+          if (element.acceptSPL) {
+            let splToken = await fetchSPLDetails(
+              cluster,
+              viewWalletPubkey,
+              element.splToken
+            )
+            if (splToken) {
```

Comment:
Looks perfect! One more thing, if the SPL token in the database is not accepted by the wallet (i.e.; `splToken` here is `null`), we should remove the SPL token from the database and set `acceptSPL` to false.

Reply:
✅